### DISCUSSION TITLE
The Mystery Box Revamp is now live!

### DISCUSSION THREAD

**Discussion by MotoLegacy** [nzp-team/nzportable#356] [Announcements]
Following the planned roadmap on my [Patreon](https://www.patreon.com/posts/hey-nz-p-heres-88294920), I spent the last couple of days rewriting almost the entirety of the Mystery Box. Here are the highlights:

## The Mystery Box now has proper leave and appear animations
Including new effects, tweaked pacing, and new sounds.

https://github.com/nzp-team/nzportable/assets/26030250/30c1b639-54da-4a13-a335-8101bc0c8ecf

## Mystery Box is now more/less likely to leave based on how many times it's used
While before the Mystery Box had a straightforward algorithm of having a 25% chance of leaving after 3 uses, it now follows a more involved algorithm to balance out obtaining an ideal loadout.

![image](https://github.com/nzp-team/nzportable/assets/26030250/15d2cddd-3e05-46c4-b2de-2e8f641dcf37)

## Use strings for obtaining Weapons from the Mystery Box now specify the Weapon
Useful for quick Weapon exchanges while in combat.

![image](https://github.com/nzp-team/nzportable/assets/26030250/b2a5a668-e354-49e5-941f-9a17a9f002df)

## New/Improved Options for Map Makers
Map Makers now have the ability to customize the following elements:
* The Mystery Box Model, and it's optional Glow
* Mystery Box Cost
* The Open and Closing sounds

With more potentially being added later. Additionally, this update brings about the new **mbox2** file format. An upgrade from the previous one, weapons are now specifically allowed/denied with strings representing their names as opposed to binary numbers, here is an example:

```
nzp_mbox2

deny:
    ray_gun
    wunderwaffe
    panzerschreck
```
Light documentation can be found in [Nacht der Untoten's MBOX2 file](https://github.com/nzp-team/assets/blob/3bb3a8e2f87a050bc27015c8cfe50393b1c424bc/common/maps/ndu.mb2).

Thanks for your support, as always.

**Comment by LexiAshe** [comment]
Awesome!!!

**Comment by DerpedCrusader** [comment]
LETS GOOOOOOOOO

finally, mp5k from box o7

**Reply by MotoLegacy** [reply]
The MP5k was already in the box

**Reply by DerpedCrusader** [reply]
Oop, guess I forgot

**Comment by BCDeshiG** [comment]
Oh sick, guess I gotta update my maps now.
